Integrations/Digital_Defense_FrontlineVM/Digital_Defense_FrontlineVM.py

```diff
@@ -0,0 +1,795 @@
+        start_time_dt = datetime.utcfromtimestamp(int(start_time))
+        start_time_str = start_time_dt.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
+        demisto.debug("FrontlineVM fetch_incident -- continuing with start_time of " + str(start_time_str))
+
+        # Fetch vulnerabilities:
+        req_params = {}
+        req_params['lte_vuln_severity_ddi'] = str(INCIDENT_VULN_MIN_SEVERITY)
+        req_params['gte_vuln_date_created'] = start_time_str
+        req_params = parse_params(req_params)
+        vulns = get_fvm_data(VULN_ENDPOINT, params=req_params)
+
+        # Create incidents
+        if vulns:
+            demisto.debug("FrontlineVM fetch_incidents -- getting all incidents.")
+            incidents = vulns_to_incident(vulns, start_time_dt)
+        else:
+            demisto.debug("FrontlineVM fetch_incidents -- no new vulnerabilities found, no incidents created.")
+            incidents = []
+
+        now = datetime.utcnow()
+        start_time = now.strftime("%s")
+        demisto.info("FrontlineVM fetch_incident -- creating new start_time of " + str(now.strftime("%Y-%m-%dT%H:%M:%S.%fZ")))
+        demisto.setLastRun({'start_time': start_time})
+        demisto.incidents(incidents)
+    except Exception as err:
+        return_error("Error: FrontlineVM fetching_incidents -- " + str(err))
+
+
+def get_hosts(ip_address, hostname, label_name, max_days_since_scan):
+    ''' Returns a list of hosts from Frontline.Cloud based on user input from Arguments    '''
+    # Prepare parameters for Frontline API request:
+    req_params = {}
+    if ip_address:
+        req_params['eq_host_ip_address'] = str(ip_address)
+    if hostname:
+        req_params['iexact_host_hostname'] = str(hostname)
+    if label_name:
+        req_params['eq_host_labels'] = str(label_name)
+    if max_days_since_scan:
+        try:
+            query_date = get_query_date_param(max_days_since_scan)
+            req_params['gte_host_date_created'] = str(query_date)
+        except ValueError:
+            debug_msg = "incorrect data type input for argument max_days_since_scan, should be number of days"
+            demisto.debug("FrontlineVM get_hosts -- " + debug_msg)
+            return_error("Error: max_days_since_scan value should be a number representing days.")
+    req_params = parse_params(req_params)
+    hosts = get_fvm_data(HOST_ENDPOINT, params=req_params)
+    return hosts
+
+
+def get_assets_command():
+    ''' Pulls host information from Frontline.Cloud '''
+    # Get Arguments:
+    ip_address = demisto.args().get('ip_address')
+    hostname = demisto.args().get('hostname')
+    label_name = demisto.args().get('label_name')
+    max_days_since_scan = demisto.args().get('max_days_since_scan')
+
+    hosts = get_hosts(ip_address, hostname, label_name, max_days_since_scan)
+
+    # Condensing Host data for HumanReadable and EntryContext:
+    asset_output = {}
+    host_list = []
+    ip_list = []
+    host_id_list = []
```

Comment:
this doesn't seem to be used anywhere

Reply:
This was used when iterating through the host data (now moved to `get_host_output(host_list)` helper function as stated previously.

Asset output is used to hold both the host_list and ip_list, and is then used as the entry context. Removed host_id_list, was not using.